crates/diman_unit_system/src/codegen/debug.rs

```diff
@@ -25,31 +25,59 @@ impl Defs {
             ..
         } = &self;
         let units = self.units_array(self.units.iter().filter(|unit| unit.factor == 1.0));
-        quote! {
-            impl<const D: #dimension_type, S: diman::DebugStorageType + std::fmt::Display> std::fmt::Debug for #quantity_type<S, D> {
-                fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-                    let closeness = |value: f64, unit_factor: f64| {
-                        if value == 0.0 {
+        if cfg!(any(feature = "std", feature = "num-traits-libm")) {
+            quote! {
+                impl<const D: #dimension_type, S: diman::DebugStorageType + core::fmt::Display> core::fmt::Debug for #quantity_type<S, D> {
+                    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+                        let closeness = |value: f64, unit_factor: f64| {
+                            if value == 0.0 {
+                                1.0
+                            } else {
+                                (value / unit_factor).abs().ln().abs()
```

Comment:
See other comment, I was just working on this on a branch, the current implementation is broken. Feel free to just comment this out for #60, if it helps. I can either fix this after your PR or after, whichever you prefer. If you need it the debug printing itself earlier, let me know.

Reply:
It was using `ln` to find the unit whose factor has the the least geometric distance to the value of the quantity, so that both 100m and 10000m get debug printed as 1km. It's just extremely ugly because of the lack of `Ord` for float and the fact that at this point, the only information about units exists in the form of this obscure array. I supposed in future versions, I'll use `integer_decode` which exists on `FloatCore` to get the exponent.

Reply:
I pushed a change to use `integer_decode` here to avoid compile breakage with those features disabled. (It doesn't fix the remaining issues.)

Reply:
If this PR looks good, I'll squash all but the first commit and update documentation of the features.

Reply:
Ok, I'll give it a look!

(I also just realized why I made the change to the Debug impl in the first place - as nice as it is to have all the SI units and prefixes generated, having `10m` printed as as `1 dam` was a little irritating, which is why I changed it to only debug print in the base unit for now. Eventually, I'll get back to this with #47  and #56, but for now I'd probably leave it as just printing it in terms of the base units.)